shell/platform/android/io/flutter/embedding/android/FlutterImageView.java

```diff
@@ -188,26 +169,23 @@ public boolean acquireLatestImage() {
     if (!isAttachedToFlutterRenderer) {
       return false;
     }
-    // There's no guarantee that the image will be closed before the next call to
-    // `acquireLatestImage()`. For example, the device may not produce new frames if
-    // it's in sleep mode, so the calls to `invalidate()` will be queued up
-    // until the device produces a new frame.
+    // 1. `acquireLatestImage()` may return null if no new image is available.
     //
-    // While the engine will also stop producing frames, there is a race condition.
+    // 2. There's no guarantee that the `onDraw()` called after `invalidate()`.
+    // For example, the device may not produce new frames if
+    // it's in sleep mode or some special Android devices so the calls to `invalidate()` will be
+    // queued up
+    // until the device produces a new frame.
     //
-    // To avoid exceptions, check if a new image can be acquired.
-    int imageOpenedCount = imageQueue.size();
-    if (currentImage != null) {
-      imageOpenedCount++;
-    }
-    if (imageOpenedCount < imageReader.getMaxImages()) {
-      final Image image = imageReader.acquireLatestImage();
-      if (image != null) {
-        imageQueue.add(image);
-      }
+    // 3. While the engine will also stop producing frames, there is a race condition.
+    final Image newImage = imageReader.acquireLatestImage();
+    if (newImage != null) {
+      // Only close current image after acquiring valid new image
+      closeCurrentImage();
```

Comment:
This potentially means that the Flutter texture is no longer in sync with the Android compositor. Let me give it a try.